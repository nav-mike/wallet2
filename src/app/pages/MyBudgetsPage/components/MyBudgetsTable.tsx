import React from 'react';
import { MyBudgetsTableHeader } from './MyBudgetsTableHeader';
import styles from './MyBudgetsTable.module.css';

export interface IBudget {
  month: string;
  year: number;
  value: number;
}

export interface IMyBudgetsTableProps {
  budgets: IBudget[];
}

export const MyBudgetsTable = (props: IMyBudgetsTableProps) => {
  return (
    <table className={styles['my-budget__table-component']}>
      <MyBudgetsTableHeader />
    </table>
  );
};
