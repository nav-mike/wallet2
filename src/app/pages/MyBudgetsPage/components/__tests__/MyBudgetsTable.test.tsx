import * as React from 'react';
import { render } from '@testing-library/react';
import { MyBudgetsTable } from '../MyBudgetsTable';

describe('<MuBudgetsTable />', () => {
  const data = [
    {
      month: 'January',
      year: 2021,
      value: 500,
    },
    {
      month: 'February',
      year: 2021,
      value: 500,
    },
  ];

  it('should render and match the snapshot', () => {
    const table = render(<MyBudgetsTable budgets={data} />);
    expect(table.container.firstChild).toMatchSnapshot();
  });
});
