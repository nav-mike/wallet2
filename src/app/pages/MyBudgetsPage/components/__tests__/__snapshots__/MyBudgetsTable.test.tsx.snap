// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuBudgetsTable /> should render and match the snapshot 1`] = `
<table
  class="my-budget__table-component"
>
  <thead>
    <tr>
      <td>
        Year
      </td>
      <td>
        Month
      </td>
      <td>
        Value
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        2021
      </td>
      <td>
        January
      </td>
      <td>
        500
      </td>
    </tr>
    <tr>
      <td>
        2021
      </td>
      <td>
        February
      </td>
      <td>
        500
      </td>
    </tr>
  </tbody>
</table>
`;
